cloudbuild-deploy.yaml

```diff
@@ -0,0 +1,34 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+steps:
+# set the project ID in the k8s manifest for service account/workload identity
+- name: 'ubuntu'
+  id: Replace
+  script: |
+    #!/usr/bin/env bash
+    sed -i "s/%GCLOUD_PROJECT%/${PROJECT_ID}/g" ./kubernetes/opentelemetry-demo.yaml
+
+# deploy the manifests to the otel-demo namespace in GKE
+- name: 'gcr.io/cloud-builders/kubectl'
+  id: Deploy
+  args:
+  - 'apply'
```

Comment:
I updated the commands in our readme for Helm, Docker, and kubectl to add `-n otel-demo` for consistency among all three. See https://github.com/GoogleCloudPlatform/opentelemetry-demo/pull/37/files:

> This also makes some updates to the Workload Identity steps to align the Helm/Kubectl install. By default, kubectl apply will install the collector in the otel-demo namespace while Helm installs in default. Instead of making more changes to the kubectl manifests or having 2 sets of instructions I just updated the commands to Helm install in the otel-demo namespace too.

This won't clean up old resources. Are there any right now that have been removed or are planning to be removed that you are aware of? If not that can be a follow up.

Reply:
I'm happy to defer cleanup to later PRs. I don't think there's anything to clean up right now, but since we're building something that should be (mostly) hands-off for months or years, we need to plan for future additions and removals to the set of k8s objects.

Are you saying that `-n otel-demo` will override the namespace of all the objects in the file? I haven't tested it but my read of the manpage is that it filters the objects, but doesn't change them.

Reply:
It's a good point to make, I copied it to https://github.com/GoogleCloudPlatform/opentelemetry-demo/issues/68 to track. Just wanted to clear up if that was a priority at this moment.

`-n` doesn't override existing namespaces set in the manifest, but not every object in the k8s yaml sets a namespace. So this puts everything in `otel-demo`. This matches the [upstream docs](https://opentelemetry.io/docs/demo/kubernetes-deployment/#install-using-kubectl), though tbh I'm not sure why upstream doesn't just set the namespace on every object in the file. Maybe something to ask there